### PR TITLE
Feature/send google api requests

### PR DESCRIPTION
**Pull request by joaoanastacio**
What is included:

- Set a default code design (folder structure mainly)
- Implementation to fetch data from Google Calendar using Google API Library

### CODE REVIEW

File: InvoiceManager/InvoiceManager/Services/Calendar/GoogleCalendarService.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+//
+
+import Foundation
+import GoogleSignIn
+import GoogleAPIClientForREST
+
+class GoogleCalendarService: CalendarServiceProtocol {
+    
+    let noItemsReturnedValue = 0
+    let requestManager = RestRequestManager.shared
```
Comment (YIshihara11201) [inline InvoiceManager/InvoiceManager/Services/Calendar/GoogleCalendarService.swift:14]: When to use this object?

File: InvoiceManager/InvoiceManager/ViewController/ImportCalendarViewController.swift [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+            presenting: self,
+            hint: nil,
+            additionalScopes: scopes
+        ) { user, error in
+            // implement handler
+            print("\(String(describing: user)) logged in")
+            
+            guard let user = GIDSignIn.sharedInstance.currentUser else { return }
+            
+            // fetch events for the last 30 days
```
Comment (YIshihara11201) [inline InvoiceManager/InvoiceManager/ViewController/ImportCalendarViewController.swift:212]: line 212 to 215 are temporary.
supposed to be changed so as to get data from data picker.

File: InvoiceManager/InvoiceManager/Services/Calendar/GoogleCalendarService.swift (hunk 1 above)
Comment (joaoanastacio) [reply InvoiceManager/InvoiceManager/Services/Calendar/GoogleCalendarService.swift:14]: This is a basic implementation based in Http request. If we need to change from Google API Library to http requests we use this request manager.

Comment (joaoanastacio) [reply InvoiceManager/InvoiceManager/Services/Calendar/GoogleCalendarService.swift:14]: In this moment we're gonna use Google API Library

File: InvoiceManager/InvoiceManager/ViewController/ImportCalendarViewController.swift (hunk 2 above)
Comment (joaoanastacio) [reply InvoiceManager/InvoiceManager/ViewController/ImportCalendarViewController.swift:212]: Yes, I haven't the implementation of date picker done before but we're gonna integrate it.

File: InvoiceManager/InvoiceManager/ViewController/ImportCalendarViewController.swift [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+                self.createDataSource()
+                //self.alert(title: "Fetched events", message: String(describing: items))
+            } else {
+                // No events
+                print("sign in before fetch!")
+            }
+        }
+    }
+    
+    func convertGoogleItemArrayToImportEvents(googleEvents: [GTLRCalendar_Event]) -> [ImportEvent]
```
Comment (YIshihara11201) [inline InvoiceManager/InvoiceManager/ViewController/ImportCalendarViewController.swift:249]: Is this functionally (& convertGoogleItemToImportEvent ) same as deserializeItemsToEvents method defined in GoogleServiceCalender.swift?

File: InvoiceManager/InvoiceManager/Services/Calendar/GoogleCalendarService.swift (hunk 1 above)
Comment (YIshihara11201) [reply InvoiceManager/InvoiceManager/Services/Calendar/GoogleCalendarService.swift:14]: OK!

**Pull request closed without merging**